src/Angor/Avalonia/Angor.Wallet/Infrastructure/WalletServices.cs

```diff
@@ -31,6 +36,7 @@ public static ServiceCollection Register(ServiceCollection services, ILogger log
         services.AddSingleton<ISensitiveWalletDataProvider, SensitiveWalletDataProvider>();
         services.AddSingleton<IWalletStore, WalletStore>();
         services.AddHttpClient();
+        services.AddSingleton<ITransactionWatcher, TransactionWatcher>();
```

Comment:
I still think there should be a specific reason to opt to singleton and otherwise we should use transient :)
lets have a talk about this when we have a meeting